hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -855,9 +855,17 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 
 	// Only update release image (triggering a new rollout) after existing rollouts
 	// have reached a terminal state.
-	rolloutComplete := hcluster.Status.Version != nil &&
-		hcluster.Status.Version.History != nil &&
-		hcluster.Status.Version.History[0].State == configv1.CompletedUpdate
+	rolloutComplete := false
+	if hcluster.Status.Version != nil && hcluster.Status.Version.History != nil {
+		currentImage := hcp.Spec.ReleaseImage
+		for _, historyItem := range hcluster.Status.Version.History {
```

Comment:
@ironcladlou @derekwaynecarr 
Do we really should be looping here? By looking at our desc for `History` the most recent update is first one in the list, so shouldn't we just check against `History[0].Image == currentImage`?

With this logic if I do the following upgrade path: a -> b -> a. Wouldn't the last upgrade against a report rolloutComplete=true (because `for _, historyItem := range hcluster.Status.Version.History {` finds the first "a" upgrade complete ) even though the current upgrade might be partial?